components/problem_tab.py

```diff
@@ -0,0 +1,255 @@
+            return table, "Display Data", "True", dash.no_update
+        figure, message, feasible, model = _solve(*args)
+        models.append(model)
+        return figure, message, feasible, models
+    return dash.no_update
+
+
+@lru_cache()
+def _generate_problem(*args):
+    ids = [setting['id'] for setting in settings] + [setting['id'] for setting
+                                                     in settings_solve]
+    kwargs = {_id: value for _id, value in zip(ids, args)}
+    kwargs['bin_length'], kwargs['bin_width'], kwargs['bin_height'] = list(
+        map(int, kwargs['bin_dimensions'].split('x'))
+    )
+    kwargs['use_cqm_solver'] = 'Constrained Quadratic Model' == kwargs['solver']
+    if kwargs['input_type'] == 'Random':
+        data = generate_data(**kwargs)
+    elif kwargs['input_type'] == 'Random Cut':
+        data = random_cut_generator(**kwargs)
+    else:
+        data = read_instance(kwargs['data_filepath'])
+    return data, kwargs
+
+
+@lru_cache()
+def _solve(*args):
+    ids = [setting['id'] for setting in settings] + [setting['id'] for setting
+                                                     in settings_solve]
+    kwargs = {_id: value for _id, value in zip(ids, args)}
+    kwargs['bin_length'], kwargs['bin_width'], kwargs['bin_height'] = list(
+        map(int, kwargs['bin_dimensions'].split('x'))
+    )
+    kwargs['use_cqm_solver'] = 'Constrained Quadratic Model' == kwargs['solver']
+    if kwargs['input_type'] == 'Random':
+        data = generate_data(**kwargs)
+    elif kwargs['input_type'] == 'Random Cut':
+        data = random_cut_generator(**kwargs)
+    else:
+        data = read_instance(kwargs['data_filepath'])
+    path = f'input/{file_name(kwargs)}.json'
+    result = main(data=data, solution_file=path, **kwargs)
+    if result['feasible']:
+        result['figure'].update_layout(
+            margin=dict(l=20, r=20, t=20, b=20),
+        )
+        kwargs['solution'] = result['solution']
+        return dcc.Graph(figure=result['figure'], style={'height': '700px'}), \
+               "Feasible Solution Found", 'success', kwargs
+    elif result['suitable']:
+        return '', \
+               "No Feasible Solution Found", 'danger', kwargs
+    else:
+        return '', "MIP cannot support quadratic interactions", 'danger', kwargs
+
+
+@app.callback(
+    [Output(f"div_{setting['id']}", 'style') for setting in settings[1:]],
+    Input('input_type', 'value'),
+    prevent_initial_callback=True
+)
+def update_inputs(value):
+    if 'Random' in value:
+        return displays
+    return [{'display': 'none' if d['display'] == 'block' or ab else 'block'}
+            for ab, d in zip(always_blocked, displays)]
+
+
+@app.callback(
+    Output("time_limit", "value"),
+    [Input("time_limit", "value"), Input("solver", "value")],
+
+)
+def adjust_time_limit(tl, solver):
+    if tl < 5 and solver == 'Constrained Quadratic Model':
+        return 5
```

Comment:
We should not silently increase a user's configuration of ``time_limit`` to use more solver time than they expect. Instead set a minimum value to the ``hybrid solver time`` field

Reply:
The minimum for CQM and CBC are not the same. The number will be updated on the screen before the problem is submitted. Maybe we can find a better solution.

Reply:
You can set the minimum value of the field based on the selected solver. It just needs a 3-line callback with the solver field as the input and the min value of the time-limit field as the output.